### PR TITLE
Validate updateWith's input parameters

### PR DESCRIPTION
**Pull request by vitaliystoliarovcc**
Validate input parameters of OrderBook's `updateWith` method. This allows to localize the issues such as appeared in [the ticket](https://app.asana.com/0/1125859137800433/1200479993218088)

Additional checks are needed in other packages that depend on this one, since it can lead to the crash of the application instead of ignoring invalid data like in the https://github.com/bitfinexcom/bfx-api-node-plugin-managed-ob/pull/6

### CODE REVIEW

**Review by tarcisiozf — APPROVED** [review]
lgtm

**Review by avsek477 — APPROVED** [review]
lgtm

File: lib/order_book.js [hunk 1]
```diff
@@ -220,6 +220,10 @@ class OrderBook extends EventEmitter {
       ? entry.length === 4 ? 2 : 1
       : 0
     const numEntry = entry.map((x) => Number(x))
+    const isValid = numEntry.every(value => Number.isFinite(value))
+
+    if (!isValid) throw new Error('the entry is not valid (should contain only numbers)')
```
Comment (vigan-abd) [inline lib/order_book.js:225]: we should return instead of throwing error here especially since this class is based on event emitter

**Review by vigan-abd — COMMENTED** [review]
please fix suggestion above

File: lib/order_book.js (hunk 1 above)
Comment (vitaliystoliarovcc) [reply lib/order_book.js:225]: @vigan-abd what should be returned if the input is invalid?

Comment (vigan-abd) [reply lib/order_book.js:225]: based on one below https://github.com/bitfinexcom/bfx-api-node-models/pull/68/files/9619439809925c1bc715d5ace30923d6f1a01be9#diff-a89d352f77a35238429009e95175b27bae98d43f1face3eddcbbf289b592ad32R262-R265 I think we should simply

**Pull request merged**